great_expectations/data_context/types/metrics.py

```diff
@@ -84,7 +84,7 @@ class NamespaceAwareExpectationDefinedValidationMetric(Metric):
         "data_asset_name": NormalizedDataAssetName,
         "batch_fingerprint": BatchFingerprint,
         "expectation_type": string_types,
-        "result_key": string_types,
+        # "result_key": tuple, # NOTE: Eugene: 2019-09-04: can we allow it to be a string or a tuple?
         "metric_kwargs": dict
```

Comment:
Can we be slightly more restrictive here to our "key friendly" dict that allows only nested versions of simple types or itself (as here) or lists of the same?

Reply:
@jcampbell Addressed - result_key is now required to be a tuple. I updated the logic so that even non-nested keys are represented as tuples, not strings.